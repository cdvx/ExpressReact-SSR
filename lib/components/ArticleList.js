import React from 'react';
import Article from './Article'

export default ({articles, articleActions})=> {
    return (
        <div>
            {Object.values(articles).map(article => 
                <Article 
                    key={article.id}
                    article={article}
                    actions={articleActions}
                />
                )}
        </div>
    )
}