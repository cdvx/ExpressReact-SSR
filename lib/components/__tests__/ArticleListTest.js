import React from 'react';
import ArticleList from '../ArticleList';

import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});
describe('ArticleList', ()=>{
    const testProps = {
        articles: {
            a: {id: 'a', "title": "a", "date": "a", "body": "a"},
            b: {id: 'b', "title": "b", "date": "b", "body": "b"}
        }
    }
    it('It renders correclty', ()=> {
        const wrapper = shallow(
            <ArticleList
            {...testProps}
            />
        )
        expect(wrapper.find('ArticleContainer').length).toEqual(2)
        expect(wrapper).toMatchSnapshot();
     })
})

