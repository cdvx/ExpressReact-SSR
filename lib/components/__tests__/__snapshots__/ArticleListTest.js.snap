// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList It renders correclty 1`] = `ShallowWrapper {}`;
